pkg/storage/replica.go

```diff
@@ -1904,17 +1905,18 @@ func (r *Replica) addWriteCmd(
 		switch retry {
 		case proposalIllegalLeaseIndex:
 			continue // retry
-		case proposalReproposed:
+		case proposalAmbiguousShouldBeReevaluated:
 			ambiguousResult = true
+			ba.WillNotBeRetried = true
 			continue // retry
 		case proposalRangeNoLongerExists, proposalErrorReproposing:
 			ambiguousResult = true
 		}
 		if pErr != nil {
-			// If this isn't an end transaction with commit=true, return
-			// error without further ado.
-			if etArg, ok := ba.GetArg(roachpb.EndTransaction); !ok ||
-				!etArg.(*roachpb.EndTransactionRequest).Commit {
+			// If this isn't a non-transactional request or an end transaction with
```

Comment:
s/isn't a non-transactional request/is a transactional request but doesn't include an end transaction/

s/`!(ok && !etArg.(*roachpb.EndTransactionRequest).Commit) && ba.Txn != nil`/`ba.Txn != nil && (!ok || !etArg.(*roachpb.EndTransactionRequest).Commit)`/